### ISSUE TITLE
Large gap in uniqueness after reproducing RL training</strong></p><p>Hi, thanks for releasing the code

### ISSUE THREAD

**Issue by Kylin-03** [wyy603/CrystalReasoner#3]
<html>
<body>
<html><head></head><body><p>I followed the README to reproduce the SFT and RL training. The SFT results are very close to the reported numbers, so the training and evaluation pipeline seem to be working correctly.</p><p>My environment:</p><ul><li><p>Python 3.12.13</p></li><li><p>PyTorch 2.8.0+cu128</p></li><li><p>CUDA 12.8</p></li><li><p>RL training on 2 × 80 GB GPUs</p></li></ul><p>The main results are:</p>

Metric | Reported SFT | Reproduced SFT | Reported RL | Reproduced RL
-- | -- | -- | -- | --
Structure validity | 91.30% | 92.21% | 94.92% | 96.05%
SMACT validity | 91.78% | 91.88% | 91.78% | 91.88%
Composition consistency | 97.25% | 98.38% | 97.32% | 98.61%
Stability | 38.04% | 38.43% | 40.95% | 39.86%
Novelty | 41.39% | 42.19% | 41.56% | 44.09%
Uniqueness | 38.91% | 39.11% | 87.23% | 39.61%
S.U.N. | 0.57% | 0.56% | 1.70% | 0.40%
Space-group consistency | 47.76% | 47.87% | 50.21% | 49.14%

<p>Most RL metrics are reasonably close to the reported results. The main difference is uniqueness: the paper reports an increase from 38.91% to 87.23%, while my reproduced result stays around 39.61%. Because of this, the S.U.N. ratio is also much lower.</p><p>I used the RL command shown in the README:</p><pre><code class="language-bash">python scripts/run.py rl_thinking_novel
</code></pre><p>Could you clarify whether this is the exact configuration used for the main RL results in the paper? I also noticed that the released checkpoint table refers to the main model as <code inline="">rl_thinking_mix</code>, so I am not sure whether I should use <code inline="">rl_thinking</code> instead.</p><p>Also, do you have any idea why the RL model in my reproduction does not improve uniqueness? Is there any additional sampling configuration, prompt construction, checkpoint selection, or evaluation setting required to obtain the reported uniqueness result?</p><p>One more issue I noticed is the stability threshold. The paper mentions <code inline="">E_hull &lt; 0.016 eV/atom</code>, while the evaluation code appears to use <code inline="">E_hull &lt; 0.1 eV/atom</code>. Could you confirm which threshold was used for the results in the paper?</p><p>I can provide the training logs, generated samples, and full evaluation outputs if needed. Thanks.</p></body></html>
</body>
</html>

**Comment by wyy603** [comment]
@Kylin-03 Hi, thanks for reproducing the paper! We rerun experiments, and conclude the increment of uniqueness is a mistake, due to data corruption during training. Moreover, for the judgement of the energy above the hull, the version in the code takes precedence.
